File: 027.RemoveElement/wyrj/RemoveElement.c
/* 0ms */
int removeElement(int* nums, int numsSize, int val) {
    int index = 0;
    while(index < numsSize) {
        if (nums[index] == val) {
            nums[index] = nums[numsSize - 1];
            numsSize--;
        } else {
            index++;
        }
    }
    return numsSize;
}

File: 027.RemoveElement/wyrj/RemoveElement.js
/* 132ms */
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function(nums, val) {
    var index = 0, len = nums.length;
    while(index < len) {
        if (nums[index] === val) {
            nums[index] = nums[len - 1];
            len--;
        } else {
            index++;
        }
    }
    return len;
};